kube-client/src/config/mod.rs

```diff
@@ -265,29 +259,21 @@ impl Config {
             root_cert = Some(ca_bundle);
         }
 
-        // REVIEW Changed behavior. This no longer fails with invalid data in PEM.
-        match loader.identity_pem() {
-            Ok(id) => identity_pem = Some(id),
-            Err(e) => {
-                tracing::debug!("failed to load client identity from kubeconfig: {}", e);
-                // last resort only if configs ask for it, and no client certs
-                if let Some(true) = loader.cluster.insecure_skip_tls_verify {
-                    accept_invalid_certs = true;
-                }
-            }
-        }
-
         Ok(Self {
             cluster_url,
             default_namespace,
             root_cert,
             timeout: Some(DEFAULT_TIMEOUT),
             accept_invalid_certs,
-            identity_pem,
             proxy_url: loader.proxy_url()?,
             auth_info: loader.user,
         })
     }
+
+    /// Client certificate and private key in PEM.
+    pub(crate) fn identity_pem(&self) -> Option<Vec<u8>> {
```

Comment:
This is only necessary at the moment because of the way the TLS functions work. I'm planning to change TLS functions to accept the certs and keys separately in the future. The identity PEM is split into a certificate chain and a private key anyway.

Having a private key separate makes it easier to implement some workarounds like converting a private key format without having to parse and extract it first. For example, #542 can be fixed by converting to PKCS#8 and that can be done in pure Rust ([topk8](https://github.com/kazk/topk8)).